tools/Az.Tools.Predictor/Az.Tools.Predictor/Predictor.cs

```diff
@@ -107,7 +92,13 @@ public Predictor(IList<string> modelPredictions, ParameterValuePredictor paramet
             // Predictions should be flexible, e.g. if "Command -Name N -Location L" is a possibility,
             // then "Command -Location L -Name N" should also be possible.
             //
-            // resultBuilder and usedParams are stored in results to help prediction more flexible.
+            // resultBuilder and usedParams are used to store the information to construct the result.
+            // We want to avoid too much heap allocation for the performance purpose.
+
+            var resultBuilder = new StringBuilder();
+            var usedParams = new HashSet<int>();
+            var presentCommands = new System.Collections.Generic.Dictionary<string, int>(); // Added
```

Comment:
Since maxAllowedCommandDupl is always 1, we don't really `presentCommand`. This can be a HashSet with the command names.
Or do you want to make `maxAllowedCommandDupl` configurable? Then you can look at Settings.cs and AzPredictorSetting.json.